dub.sdl

```diff
@@ -94,3 +117,54 @@ subPackage {
 
   dependency "dmd:parser" version="*"
 }
+
+/* The 'full' subpackage includes frontend + backend*/
+subPackage {
+  name "full"
+
+  targetType "library"
+  sourcePaths "src/dmd"
+  stringImportPaths "res"
+  stringImportPaths "generated/dub"
+
+  versions \
+    "GC" \
+    "NoMain" \
+    "MARS"
+
+  preBuildCommands `
+    "$${DUB_EXE}" \
+    --arch=$${DUB_ARCH} \
+    --compiler=$${DC} \
+    --single "$${DUB_PACKAGE_DIR}config.d" \
+    -- "$${DUB_PACKAGE_DIR}generated/dub" \
+    "$${DUB_PACKAGE_DIR}VERSION" \
+    /etc
+   ` platform="posix"
+
+  preBuildCommands `"%DUB_EXE%" --arch=%DUB_ARCH% --compiler="%DC%" --single "%DUB_PACKAGE_DIR%config.d" -- "%DUB_PACKAGE_DIR%generated/dub" "%DUB_PACKAGE_DIR%VERSION"` platform="windows"
+}
+
+configuration "application" {
```

Comment:
Can this depend on `full` instead? Seems more logical.

Reply:
Yes, logically, but practically no, because `full` uses the `NoMain` version (library build).

Reply:
Ah, I see.

Reply:
Perhaps the code that is used when `NoMain` is **not** defined should be moved to its own file.

Reply:
Yes, but I couldn't get that to work as dub seems to add `sourcePaths = ["src"]` to all configurations by default and doesn't allow it to be removed. Unfortunately, `excludedSourceFiles` has the highest priority, so we can't just exclude everything and then add a single file. It only works for the library config, because of the `none` targetType.

https://dub.pm/package-format-sdl.html#target-types

Reply:
Hmm, ok.